LINQ.Exercises/Orderby.cs

```diff
@@ -0,0 +1,217 @@
+
+            Assert.IsTrue(result.SequenceEqual(new string[] { "apple", "zuchini", "blueberry", "cherry", "tamarind" }));
+        }
+
+        [TestMethod]
+        public void OrderDoublesFromLargestToSmallest_returnDescendingEnumeration()
+        {
+            double[] doubles = { 1.7, 2.3, 1.9, 4.1, 2.9 };
+
+            IEnumerable<double> result = doubles.OrderByDescending(number => number);
+
+            Assert.IsTrue(result.SequenceEqual(new double[] { 4.1, 2.9, 2.3, 1.9, 1.7 }));
+        }
+
+        [TestMethod]
+        public void OrderByBirthDatesOldestToYoungest_returnPersonEnumeration()
+        {
+            IEnumerable<TestData.Person> result = TestData.People.OrderBy(person => person.Born.Date);
+
+            Assert.IsTrue(result.SequenceEqual(new TestData.Person[] {
+                    new TestData.Person("Jean", "Gean", new DateTime(1950, 12, 1)),
+                    new TestData.Person("Jimmy", "Jilly", new DateTime(1974, 9, 16)),
+                    new TestData.Person("Jack", "Tuck", new DateTime(1990, 3, 12)),
+                    new TestData.Person("Jill", "Lill", new DateTime(2001, 5, 21))
+            }));
+        }
+
+        [TestMethod]
+        public void OrderByBirthDatesYoungestToOldest_returnPersonEnumeration()
+        {
+            IEnumerable<TestData.Person> result = TestData.People.OrderByDescending(person => person.Born.Date);
+
+            Assert.IsTrue(result.SequenceEqual(new TestData.Person[] {
+                    new TestData.Person("Jill", "Lill", new DateTime(2001, 5, 21)),
+                    new TestData.Person("Jack", "Tuck", new DateTime(1990, 3, 12)),
+                    new TestData.Person("Jimmy", "Jilly", new DateTime(1974, 9, 16)),
+                    new TestData.Person("Jean", "Gean", new DateTime(1950, 12, 1))
+            }));
+        }
+
+        [TestMethod]
+        public void OrderByLengthAndThenAlphabetically_returnStringNumberation()
+        {
+            string[] digits = { "zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine" };
+            IEnumerable<string> result = digits.OrderBy(digit => digit.Length).ThenBy(digit => digit);
+
+            Assert.IsTrue(result.SequenceEqual(new string[] { "one", "six", "two", "five", "four", "nine", "zero", "eight", "seven", "three" }));
+        }
+
+        // order by increasing length
+        // then by alphabetically from a to z, ignoring case
+        [TestMethod]
+        public void OrderByLengthThenAlphabeticallyIgnoringCase_returnStringEnumeration()
+        {
+            #region hint
+
+            // you will have to create a class which implements the IComparer<string> interface
+
+            #endregion hint
+
+            string[] words = { "aPPLE", "AbAcUs", "bRaNcH", "BlUeBeRrY", "ClOvEr", "cHeRry" };
+
+            IEnumerable<string> result = words.OrderBy(word => word.Length).ThenBy(word => word);
```

Comment:
please implement your own `Comparer<string>`

Reply:
Will be done